npc/analyser.go

```diff
@@ -50,6 +51,14 @@ func (ns *ns) analysePolicy(policy *networkingv1.NetworkPolicy) (
 			// If From is empty or missing, this rule matches all sources
 			allSources := ingressRule.From == nil || len(ingressRule.From) == 0
 
+			if !allPorts {
+				for _, npProtocolPort := range ingressRule.Ports {
```

Comment:
Could we create a function which would return the error if a named port is found among given ports? In this way, we could avoid the duplication.

Reply:
Done